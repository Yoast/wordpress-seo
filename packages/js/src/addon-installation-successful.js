/* global wpseoAddonInstallationL10n */

import { render } from "@wordpress/element";
import AddonInstallationSuccessful from "./components/AddonInstallationSuccessful";

const elementToInsert = document.createElement( "div" );
elementToInsert.setAttribute( "id", "wpseo-app-addon-installation-success-message" );

// Insert under the first heading on the page.
document.getElementById( "yoast-heading-extensions" ).append( elementToInsert );

render(
	<AddonInstallationSuccessful configurationWizardUrl={ wpseoAddonInstallationL10n.configurationWizardUrl } />,
	elementToInsert
);
