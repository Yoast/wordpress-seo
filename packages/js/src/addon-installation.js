/* global wpseoAddonInstallationL10n */

import { render } from "@wordpress/element";
import AddonInstallation from "./components/AddonInstallation";

const element = document.createElement( "div" );
element.setAttribute( "id", "wpseo-app-element" );
document.getElementById( "extensions" ).append( element );

render(
	<AddonInstallation nonce={ wpseoAddonInstallationL10n.nonce } addons={ wpseoAddonInstallationL10n.addons } />,
	element
);
