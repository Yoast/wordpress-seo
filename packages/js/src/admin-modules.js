import CornerstoneWorkoutCard from "./workouts/components/CornerstoneWorkoutCard";
import OrphanedWorkoutCard from "./workouts/components/OrphanedWorkoutCard";

window.yoast = window.yoast || {};
window.yoast.adminModules = {
	components: {
		workouts: {
			CornerstoneWorkoutCard,
			OrphanedWorkoutCard,
		},
	},
};
