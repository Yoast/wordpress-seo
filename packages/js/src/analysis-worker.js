// Ensure the global window is set, our dependencies use it.
self.window = self;

/**
 * These dependencies reference objects and methods that
 * are not available inside a web worker, so we disallow
 * loading them.
 *
 * We need to do that in this file, since we have no
 * control over the dependencies of WordPress dependencies that we want to load.
 *
 * @type {string[]}
 */
const disallowedDependencies = [
	/**
	 * References `Element`, a DOM interface not available in web workers.
	 *
	 * Was renamed, hence the two variants.
	 */
	"wp-inert-polyfill",
	"wp-polyfill-inert",
];

/**
 * Loads dependencies.
 *
 * @param {Object<string, string>} dependencies The dependencies.
 *
 * @returns {void}
 */
function loadDependencies( dependencies ) {
	for ( const dependency in dependencies ) {
		if ( ! Object.prototype.hasOwnProperty.call( dependencies, dependency ) ) {
			continue;
		}

		if ( disallowedDependencies.includes( dependency ) ) {
			continue;
		}

		self.importScripts( dependencies[ dependency ] );

		if ( dependency === "lodash" ) {
			// eslint-disable-next-line no-undef
	        self.lodash = _.noConflict();
		}
	}
}

/**
 * Loads translations.
 *
 * @param {Object[]} translations The translations.
 *
 * @returns {void}
 */
function loadTranslations( translations ) {
	for ( const [ domain, translation ] of translations ) {
		var localeData = translation.locale_data[ domain ] || translation.locale_data.messages;
		localeData[ "" ].domain = domain;
		self.wp.i18n.setLocaleData( localeData, domain );
	}
}

self.onmessage = ( { data } ) => {
	if ( ! data || ! data.dependencies ) {
		return;
	}

	loadDependencies( data.dependencies );

	if ( data.translations ) {
		loadTranslations( data.translations );
	}

	const Researcher = self.yoast.Researcher.default;
	const worker = new self.yoast.analysis.AnalysisWebWorker( self, new Researcher() );
	worker.register();
};
