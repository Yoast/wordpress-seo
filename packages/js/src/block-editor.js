/* eslint-disable no-unused-vars */
import initBlockEditorIntegration from "./initializers/block-editor-integration";
import BlockEditorData from "./analysis/blockEditorData";

window.yoast = window.yoast || {};
window.yoast.initEditorIntegration = initBlockEditorIntegration;
window.yoast.EditorData = BlockEditorData;
/* eslint-enable no-unused-vars */
