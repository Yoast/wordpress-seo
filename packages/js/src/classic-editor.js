/* eslint-disable no-unused-vars */
import initClassicEditorIntegration from "./initializers/classic-editor-integration";
import ClassicEditorData from "./analysis/classicEditorData";

window.yoast = window.yoast || {};
window.yoast.initEditorIntegration = initClassicEditorIntegration;
window.yoast.EditorData = ClassicEditorData;
/* eslint-enable no-unused-vars */
