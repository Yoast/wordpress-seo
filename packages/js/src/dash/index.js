export { Dashboard } from "./components/dashboard";
