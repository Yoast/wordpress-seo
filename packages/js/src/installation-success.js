import domReady from "@wordpress/dom-ready";
import { __, sprintf } from "@wordpress/i18n";
import { createInterpolateElement } from "@wordpress/element";
import { renderReactRoot } from "./helpers/reactRoot";
import { CheckIcon } from "@heroicons/react/solid";
import { ArrowRightIcon } from "@heroicons/react/outline";

/**
 * The installation success page.
 *
 * @returns {WPElement} The installation success page.
 */
function InstallationSuccessPage() {
	return (
		<div className="yst-root yst-my-auto yst-flex yst-flex-col yst-h-[90vh] yst-justify-center">
			<h1 className="yst-text-4xl yst-text-gray-900 yst-w-[350px] yst-font-extrabold yst-leading-10 yst-mx-auto yst-mt-6 yst-mb-16 yst-text-center yst-tracking-tight">
				{
					createInterpolateElement(
						sprintf(
							/* translators: %s expands to Yoast SEO. */
							__( "You've successfully installed %s!", "wordpress-seo" ),
							// The space between 'Yoast' and 'SEO' is a non-breaking space (Unicode 160).
							"<span>Yoast SEO</span>" ),
						{
							span: <span className="yst-text-primary-500" />,
						}
					)
				}
			</h1>
			<div className="installation-success-content">
				<div className="yst-flex yst-flex-col md:yst-flex-row yst-justify-center yst-items-center yst-gap-8">
					<div id="installation-success-card-optimized-site" className="yst-shrink-0 yst-bg-white yst-rounded-lg yst-p-6 yst-flex yst-flex-col yst-max-w-sm yst-shadow-md yst-h-4/5 yst-leading-6">
						<h2 className="yst-tracking-tight yst-text-gray-900 yst-text-2xl yst-leading-8 yst-font-extrabold">{ __( "Your site is already easier to find for search engines.", "wordpress-seo" ) }</h2>
						<p className="yst-text-gray-500 yst-text-base yst-my-4">
							{ sprintf(
								/* translators: %s expands to Yoast SEO. */
								__( "%s rolls out the red carpet for the search bots, which helps your site perform better in search engines.",
									"wordpress-seo" ),
								"Yoast SEO"
							) }
						</p>
						<div className="card-button-section" />
						<div className="yst-bg-green-100 yst-w-20 yst-h-20 yst-rounded-full yst-mx-auto yst-my-2 yst-flex yst-items-center yst-justify-center">
							<CheckIcon className="yst-w-8 yst-text-green-600" />
						</div>
					</div>
					<div>
						<ArrowRightIcon className="yst-w-8 yst-text-gray-500" />
					</div>
					<div id="installation-success-card-configuration" className="yst-shrink-0 yst-shadow-xl yst-bg-primary-500 yst-rounded-lg yst-p-6 yst-flex yst-flex-col yst-max-w-sm yst-h-4/5 yst-leading-6">
						<h2 className=" yst-text-white yst-text-2xl yst-leading-8 yst-font-extrabold">
							{ sprintf(
								/* translators: %s expands to Yoast SEO. */
								__( "For the best ranking results: configure %s!", "wordpress-seo" ),
								"Yoast SEO"
							) }
						</h2>
						<p className="yst-font-medium yst-text-white yst-text-base yst-my-4">
							{ sprintf(
								/* translators: %s expands to Yoast SEO. */
								__( "Set the essential %s settings for your site in a few steps.", "wordpress-seo" ),
								"Yoast SEO"
							) }
						</p>
						<img
							className="yst-my-6 yst-mx-auto yst-w-[150px] yst-h-[120px]"
							src={ window.wpseoInstallationSuccess.pluginUrl + "/images/indexables_3_left_bubble_optm.svg" }
							alt={ "" }
						/>
						<div className="yst-flex yst-grow-1 yst-mt-auto">
							<a
								id="installation-successful-configuration-link"
								href={ window.wpseoInstallationSuccess.firstTimeConfigurationUrl }
								className="yst-inline-flex yst-items-center yst-w-full yst-justify-center yst-no-underline yst-px-6 yst-py-3 yst-border yst-border-transparent yst-text-base yst-font-medium yst-rounded-md yst-shadow-none yst-text-primary-500 yst-bg-white hover:yst-bg-gray-50 focus:yst-outline-none focus:yst-ring-2 focus:yst-ring-offset-2 focus:yst-ring-white yst-ring-offset-2 yst-ring-offset-primary-500"
								data-hiive-event-name="clicked_start_first_time_configuration"
							>
								{ __( "Start first-time configuration!", "wordpress-seo" ) }
								<svg
									xmlns="http://www.w3.org/2000/svg"
									className="yst-ml-3 yst-mr-1 yst-h-5 yst-w-5"
									fill="none"
									viewBox="0 0 24 24"
									stroke="currentColor"
								>
									<path
										strokeLinecap="round"
										strokeLinejoin="round"
										strokeWidth="2"
										d="M14 5l7 7m0 0l-7 7m7-7H3"
									/>
								</svg>
							</a>
						</div>
					</div>
					<a
						id="installation-success-skip-link"
						className="yst-bottom-12 yst-right-0 yst-mr-5 yst-self-end yst-text-base md:yst-absolute"
						href={ "/wp-admin/admin.php?page=wpseo_dashboard" }
						data-hiive-event-name="clicked_skip_button | installation successful screen"
					>
						{
							/* translators: %s expands to ' »'. */
							sprintf( __( "Skip%s", "wordpress-seo" ), " »" )
						}
					</a>
				</div>
			</div>
		</div>
	);
}

domReady( () => {
	renderReactRoot( "wpseo-installation-successful-free", <InstallationSuccessPage /> );
} );
