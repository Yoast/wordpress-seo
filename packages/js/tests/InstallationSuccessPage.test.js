import { InstallationSuccessPage } from "../src/installation-success";
import { render } from "./test-utils";

describe( "InstallationSuccessPage", () => {
	it( "should render the text 'Your site is now easier for search engines to find.'", () => {
		global.window.wpseoInstallationSuccess = {
			pluginUrl: "http://wordpress.test/wp-content/plugins/wordpress-seo",
			firstTimeConfigurationUrl: "http://wordpress.test/wp-admin/admin.php?page=wpseo_dashboard#/first-time-configuration",
			dashboardUrl: "http://wordpress.test/wp-admin/admin.php?page=wpseo_dashboard",
		};

		const { queryByText } = render( <InstallationSuccessPage /> );
		expect( queryByText( "Your site is now easier for search engines to find." ) ).toBeInTheDocument();
	} );
	it( "should match the snapshot", () => {
		const { container } = render( <InstallationSuccessPage /> );
		expect( container ).toMatchSnapshot();
	} );
} );
