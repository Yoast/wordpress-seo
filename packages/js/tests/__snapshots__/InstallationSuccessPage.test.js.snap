// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstallationSuccessPage should match the snapshot 1`] = `
<div>
  <div
    class="yst-root yst-my-auto yst-flex yst-flex-col yst-h-[90vh] yst-justify-center"
  >
    <h1
      class="yst-text-4xl yst-text-gray-900 yst-w-[350px] yst-font-extrabold yst-leading-10 yst-mx-auto yst-mt-6 yst-mb-16 yst-text-center yst-tracking-tight"
    >
      You've successfully installed 
      <span
        class="yst-text-primary-500"
      >
        Yoast SEO
      </span>
      !
    </h1>
    <div
      class="installation-success-content"
    >
      <div
        class="yst-flex yst-flex-col md:yst-flex-row yst-justify-center yst-items-center yst-gap-8"
      >
        <div
          class="yst-shrink-0 yst-bg-white yst-rounded-lg yst-p-6 yst-flex yst-flex-col yst-max-w-sm yst-shadow-md yst-h-4/5 yst-leading-6"
          id="installation-success-card-optimized-site"
        >
          <h2
            class="yst-tracking-tight yst-text-gray-900 yst-text-2xl yst-leading-8 yst-font-extrabold"
          >
            Your site is now easier for search engines to find.
          </h2>
          <p
            class="yst-text-gray-500 yst-text-base yst-my-4"
          >
            Yoast SEO rolls out the red carpet for the search bots, which helps your site perform better in search engines.
          </p>
          <div
            class="card-button-section"
          />
          <div
            class="yst-bg-green-100 yst-w-20 yst-h-20 yst-rounded-full yst-mx-auto yst-my-2 yst-flex yst-items-center yst-justify-center"
          >
            <svg
              aria-hidden="true"
              class="yst-w-8 yst-text-green-600"
              fill="currentColor"
              viewBox="0 0 20 20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M16.707 5.293a1 1 0 010 1.414l-8 8a1 1 0 01-1.414 0l-4-4a1 1 0 011.414-1.414L8 12.586l7.293-7.293a1 1 0 011.414 0z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
        <div>
          <svg
            aria-hidden="true"
            class="yst-w-8 yst-text-gray-500 rtl:yst-rotate-180"
            fill="none"
            stroke="currentColor"
            stroke-width="2"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 5l7 7m0 0l-7 7m7-7H3"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
        <div
          class="yst-shrink-0 yst-shadow-xl yst-bg-primary-500 yst-rounded-lg yst-p-6 yst-flex yst-flex-col yst-max-w-sm yst-h-4/5 yst-leading-6"
          id="installation-success-card-configuration"
        >
          <h2
            class=" yst-text-white yst-text-2xl yst-leading-8 yst-font-extrabold"
          >
            For the best ranking results: configure Yoast SEO!
          </h2>
          <p
            class="yst-font-medium yst-text-white yst-text-base yst-my-4"
          >
            Set the essential Yoast SEO settings for your site in a few steps.
          </p>
          <img
            alt=""
            class="yst-my-6 yst-mx-auto yst-w-[150px] yst-h-[120px]"
            src="http://wordpress.test/wp-content/plugins/wordpress-seo/images/indexables_3_left_bubble_optm.svg"
          />
          <div
            class="yst-flex yst-grow-1 yst-mt-auto"
          >
            <a
              class="yst-inline-flex yst-items-center yst-w-full yst-justify-center yst-no-underline yst-px-6 yst-py-3 yst-border yst-border-transparent yst-text-base yst-font-medium yst-rounded-md yst-shadow-none yst-text-primary-500 yst-bg-white hover:yst-bg-gray-50 focus:yst-outline-none focus:yst-ring-2 focus:yst-ring-offset-2 focus:yst-ring-white yst-ring-offset-2 yst-ring-offset-primary-500"
              data-hiive-event-name="clicked_start_first_time_configuration"
              href="http://wordpress.test/wp-admin/admin.php?page=wpseo_dashboard#/first-time-configuration"
              id="installation-successful-configuration-link"
            >
              Start first-time configuration!
              <svg
                class="yst-ms-3 yst-me-1 yst-h-5 yst-w-5 rtl:yst-rotate-180"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14 5l7 7m0 0l-7 7m7-7H3"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </a>
          </div>
        </div>
        <a
          class="yst-bottom-12 yst-right-0 yst-mr-5 yst-self-end yst-text-base md:yst-absolute"
          data-hiive-event-name="clicked_skip_button | installation successful screen"
          href="http://wordpress.test/wp-admin/admin.php?page=wpseo_dashboard"
          id="installation-success-skip-link"
        >
          Skip »
        </a>
      </div>
    </div>
  </div>
</div>
`;
